java/src/main/java/com/ibm/plugin/rules/detection/bc/mac/BcMac.java

```diff
@@ -203,9 +203,10 @@ private BcMac() {
                             .createDetectionRule()
                             .forObjectTypes("org.bouncycastle.crypto.macs." + macClass)
                             .forConstructor()
-                            .shouldBeDetectedAs(new ValueActionFactory<>(macClass))
+                            //     .shouldBeDetectedAs(new ValueActionFactory<>(macClass))
                             .withMethodParameter("org.bouncycastle.crypto.Digest")
-                            .addDependingDetectionRules(BcDigests.rules())
+                            .addDependingDetectionRules(
+                                    BcDigests.rules(new MacContext(MacContext.Kind.HMAC)))
```

Comment:
Is there a reason why the Kind is still used here when it is already deprecated? [see](Add the `@Deprected` annotation similar to [here](https://github.com/IBM/sonar-cryptography/blob/main/engine/src/main/java/com/ibm/engine/model/context/CipherContext.java#L61-L70))

Reply:
I just needed this to make it work quickly, but I plan on removing the use of `Kind` in the entire BouncyCastle transaction in a later time, once I finish making everything work.